models/issues/pull_list.go

```diff
@@ -199,3 +199,14 @@ func (prs PullRequestList) GetIssueIDs() []int64 {
 	}
 	return issueIDs
 }
+
+// CountMergedPullRequestInRepo return the count the user merged into the repository via pull request
+func CountMergedPullRequestInRepo(ctx context.Context, repoID, posterID int64) (int64, error) {
+	return db.GetEngine(ctx).
+		Join("INNER", "pull", "pull.issue_id = issue.id").
+		Where("repo_id=?", repoID).
+		And("poster_id=?", posterID).
+		And("is_pull=?", true).
+		And("pull.has_merged=?", true).
+		Count(new(Issue))
```

Comment:
@lunny it could be optimized to `HasMergedPullRequestInRepo: limit 1`, then database stops at the first matched record, it's faster.